packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -40,6 +40,41 @@ import {
   parseCaipChainId,
 } from '@metamask/utils';
 
+/**
+ * A list of stateful component types.
+ */
+const STATEFUL_COMPONENT_TYPES = [
+  'Input',
+  'Dropdown',
+  'RadioGroup',
+  'FileInput',
+  'Checkbox',
+  'Selector',
+  'AssetSelector',
+  'AddressInput',
+] as const;
+
+/**
+ * Type for stateful component types.
+ */
+type StatefulComponentType = (typeof STATEFUL_COMPONENT_TYPES)[number];
+
+/**
+ * Check if a component is a stateful component.
+ *
+ * @param component - The component to check.
+ * @param component.type - The type of the component.
+ *
+ * @returns Whether the component is a stateful component.
+ */
+export function isStatefulComponent(component: { type: string }): component is {
+  type: StatefulComponentType;
```

Comment:
Oh, does this then properly infer the props inside the if-clause in `constructState`? 🤔

Reply:
Yes, I have no issues in my IDE and when I look at the infered type its an union of the stateful components, I would like to send you a screenshot but I can't hit `ctrl` without the IDE context widow closing :sweat_smile:

Reply:
Awesome! 👍